hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,574 @@
+                        "label": "Show Conflicts",
+                        "tooltip": "Show the conflicting publishes in Shotgun",
+                        "text": conflict_info
+                    }
+                }
+            )
+
+        # store the session path since this is guaranteed to run just before
+        # the publish itself
+        item.properties["publish_file_path"] = path
+
+        # store the publish info in the properties
+        item.properties["publish_info"] = publish_info
+
+        self.logger.info("A Publish will be created in Shotgun and linked to:")
+        self.logger.info("  %s" % (path,))
+
+        return True
+
+    def publish(self, settings, item):
+        """
+        Executes the publish logic for the given item and settings.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        publisher = self.parent
+
+        # get the work file path stored during validation
+        path = item.properties["publish_file_path"]
+
+        # get all the publish info extracted during validation
+        publish_info = item.properties["publish_info"]
+        version_number = publish_info["publish_version"]
+        publish_path = publish_info["publish_path"]
+        publish_name = publish_info["publish_name"]
+        publish_type = publish_info["publish_type"]
+
+        # set the alembic args that make the most sense when working with Mari.  These flags
+        # will ensure the export of an Alembic file that contains all visible geometry from
+        # the current scene together with UV's and face sets for use in Mari.
+        alembic_args = ["-renderableOnly",   # only renderable objects (visible and not templated)
+                        "-writeFaceSets",    # write shading group set assignments (Maya 2015+)
+                        "-uvWrite"           # write uv's (only the current uv set gets written)
+                        ]        
+
+        # find the animated frame range to use:
+        start_frame, end_frame = self._find_scene_animation_range()
+        if start_frame and end_frame:
+            alembic_args.append("-fr %d %d" % (start_frame, end_frame))
+
+        # Set the output path: 
+        # Note: The AbcExport command expects forward slashes!
+        alembic_args.append("-file %s" % publish_path.replace("\\", "/"))
+
+        # build the export command.  Note, use AbcExport -help in Maya for
+        # more detailed Alembic export help
+        abc_export_cmd = ("AbcExport -j \"%s\"" % " ".join(alembic_args))
+
+        # ...and execute it:
+        try:
+            self.parent.log_debug("Executing command: %s" % abc_export_cmd)
+            mel.eval(abc_export_cmd)
+        except Exception, e:
+            raise TankError("Failed to export Alembic Cache: %s" % e)
```

Comment:
We should log an error here and return instead of raising.

Reply:
logged error and returned